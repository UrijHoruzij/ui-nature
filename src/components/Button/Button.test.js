import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { Button, Icon } from "../";
import ThemeProvider from "../../ThemeProvide";

describe("Button component", () => {
  test("Render button", () => {
    const component = (
      <ThemeProvider>
        <Button type="primary">Primary</Button>
        <Button type="secondary">Secondary</Button>
        <Button type="outline">Outline</Button>
        <Button type="outline-primary">Outline-primary</Button>
        <Button type="link">Link</Button>
        <Button type="link-primary">Link-primary</Button>
        <Button type="primary" disabled>
          Disabled
        </Button>
        <Button type="primary" rounded>
          Войти
        </Button>
        <Button type="primary" fullWidth>
          Войти
        </Button>
        <Button
          iconPosition="left"
          icon={<Icon name="close" color="contrast" />}
          type="primary"
        >
          Закрыть
        </Button>
        ;
        <Button
          iconPosition="right"
          icon={<Icon name="close" color="contrast" />}
          type="primary"
        >
          Закрыть
        </Button>
        ;
        <Button
          overlay={<input type="file" />}
          icon={<Icon name="close" iconPosition="right" color="contrast" />}
          type="primary"
        >
          Закрыть
        </Button>
        ;
        <Button loadingData type="primary">
          Закрыть
        </Button>
      </ThemeProvider>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
