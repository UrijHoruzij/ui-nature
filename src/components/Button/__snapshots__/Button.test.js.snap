// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Render button 1`] = `
Array [
  .c0 {
  font-family: Montserrat,sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  margin: 0;
  padding: 8px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 32px;
  border: none;
  border-radius: 4px;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
  box-sizing: border-box;
  color: #ffffff;
  background-color: #008603;
}

.c0:hover {
  background-color: #339e35;
}

<button
    className="c0"
    type="primary"
  >
    Primary
  </button>,
  .c0 {
  font-family: Montserrat,sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  margin: 0;
  padding: 8px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 32px;
  border: none;
  border-radius: 4px;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
  box-sizing: border-box;
  color: #333333;
  background-color: #f0f0f0;
}

.c0:hover {
  background-color: #b5b5b5;
}

<button
    className="c0"
    type="secondary"
  >
    Secondary
  </button>,
  .c0 {
  font-family: Montserrat,sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  margin: 0;
  padding: 8px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 32px;
  border: none;
  border-radius: 4px;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
  box-sizing: border-box;
  color: #333333;
  background-color: transparent;
  border: 1px solid #b5b5b5;
}

.c0:hover {
  background-color: #f0f0f0;
}

<button
    className="c0"
    type="outline"
  >
    Outline
  </button>,
  .c0 {
  font-family: Montserrat,sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  margin: 0;
  padding: 8px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 32px;
  border: none;
  border-radius: 4px;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
  box-sizing: border-box;
  color: #333333;
  background-color: transparent;
  border: 1px solid #008603;
}

.c0:hover {
  background-color: #008603;
  color: #ffffff;
}

<button
    className="c0"
    type="outline-primary"
  >
    Outline-primary
  </button>,
  .c0 {
  font-family: Montserrat,sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  margin: 0;
  padding: 8px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 32px;
  border: none;
  border-radius: 4px;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
  box-sizing: border-box;
  padding: 0;
  height: 100%;
  color: #333333;
  background-color: transparent;
}

.c0:hover {
  color: #333333;
}

<button
    className="c0"
    type="link"
  >
    Link
  </button>,
  .c0 {
  font-family: Montserrat,sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  margin: 0;
  padding: 8px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 32px;
  border: none;
  border-radius: 4px;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
  box-sizing: border-box;
  padding: 0;
  height: 100%;
  color: #008603;
  background-color: transparent;
}

.c0:hover {
  color: #339e35;
}

<button
    className="c0"
    type="link-primary"
  >
    Link-primary
  </button>,
  .c0 {
  font-family: Montserrat,sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  margin: 0;
  padding: 8px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 32px;
  border: none;
  border-radius: 4px;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: default;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
  box-sizing: border-box;
  color: #ffffff;
  background-color: #008603;
  color: #b5b5b5;
  background-color: #f0f0f0;
}

.c0:hover {
  background-color: #339e35;
}

.c0:hover {
  color: #b5b5b5;
  background-color: #f0f0f0;
}

<button
    className="c0"
    disabled={true}
    type="primary"
  >
    Disabled
  </button>,
  .c0 {
  font-family: Montserrat,sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  margin: 0;
  padding: 8px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 32px;
  border: none;
  border-radius: 4px;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
  box-sizing: border-box;
  color: #ffffff;
  background-color: #008603;
  border-radius: 32px;
}

.c0:hover {
  background-color: #339e35;
}

<button
    className="c0"
    type="primary"
  >
    Войти
  </button>,
  .c0 {
  font-family: Montserrat,sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  margin: 0;
  padding: 8px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 32px;
  border: none;
  border-radius: 4px;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
  box-sizing: border-box;
  color: #ffffff;
  background-color: #008603;
  width: 100%;
}

.c0:hover {
  background-color: #339e35;
}

<button
    className="c0"
    type="primary"
  >
    Войти
  </button>,
  .c0 {
  font-family: Montserrat,sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  margin: 0;
  padding: 8px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 32px;
  border: none;
  border-radius: 4px;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
  box-sizing: border-box;
  color: #ffffff;
  background-color: #008603;
}

.c0:hover {
  background-color: #339e35;
}

.c1 {
  width: 16px;
  height: 16px;
  margin-right: 4px;
}

.c2 {
  stroke: #ffffff;
  -webkit-transition: all .3s ease-in-out;
  transition: all .3s ease-in-out;
}

<button
    className="c0"
    type="primary"
  >
    <div
      className="c1"
    >
      <svg
        className="c2"
        color="contrast"
        fill="none"
        height="16"
        name="close"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 2L14 14"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
        <path
          d="M14 2L2 14"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
    </div>
    Закрыть
  </button>,
  ";",
  .c0 {
  font-family: Montserrat,sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  margin: 0;
  padding: 8px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 32px;
  border: none;
  border-radius: 4px;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
  box-sizing: border-box;
  color: #ffffff;
  background-color: #008603;
}

.c0:hover {
  background-color: #339e35;
}

.c1 {
  width: 16px;
  height: 16px;
  margin-left: 4px;
}

.c2 {
  stroke: #ffffff;
  -webkit-transition: all .3s ease-in-out;
  transition: all .3s ease-in-out;
}

<button
    className="c0"
    type="primary"
  >
    Закрыть
    <div
      className="c1"
    >
      <svg
        className="c2"
        color="contrast"
        fill="none"
        height="16"
        name="close"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 2L14 14"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
        <path
          d="M14 2L2 14"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
    </div>
  </button>,
  ";",
  .c0 {
  font-family: Montserrat,sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  margin: 0;
  padding: 8px 0px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 32px;
  border: none;
  border-radius: 4px;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
  box-sizing: border-box;
  color: #ffffff;
  background-color: #008603;
}

.c0:hover {
  background-color: #339e35;
}

.c4 {
  width: 100%;
  height: 100%;
  position: absolute;
  outline: none;
  opacity: 0;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  width: 100%;
  height: 100%;
  padding: 8px 16px;
}

.c2 {
  width: 16px;
  height: 16px;
  margin-right: 4px;
}

.c3 {
  stroke: #ffffff;
  -webkit-transition: all .3s ease-in-out;
  transition: all .3s ease-in-out;
}

<button
    className="c0"
    type="primary"
  >
    <label
      className="c1"
    >
      <div
        className="c2"
      >
        <svg
          className="c3"
          color="contrast"
          fill="none"
          height="16"
          name="close"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 2L14 14"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
          <path
            d="M14 2L2 14"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
      Закрыть
      <span
        className="c4"
      >
        <input
          type="file"
        />
      </span>
    </label>
  </button>,
  ";",
  .c0 {
  font-family: Montserrat,sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  margin: 0;
  padding: 8px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 32px;
  border: none;
  border-radius: 4px;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
  box-sizing: border-box;
  color: #ffffff;
  background-color: #008603;
}

.c0:hover {
  background-color: #339e35;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  width: 4px;
  height: 4px;
  border-radius: 4px;
  background-color: #ffffff;
  -webkit-animation: jviYpE 1s infinite;
  animation: jviYpE 1s infinite;
}

.c2:nth-child(1) {
  margin-right: 4px;
  -webkit-animation-delay: 0.33s;
  animation-delay: 0.33s;
}

.c2:nth-child(2) {
  -webkit-animation-delay: 0.66s;
  animation-delay: 0.66s;
}

.c2:nth-child(3) {
  margin-left: 4px;
  -webkit-animation-delay: 0.99s;
  animation-delay: 0.99s;
}

<button
    className="c0"
    type="primary"
  >
    <span
      className="c1"
    >
      <span
        className="c2"
      />
      <span
        className="c2"
      />
      <span
        className="c2"
      />
    </span>
  </button>,
]
`;
